app/controllers/match_contacts_controller.rb

```diff
@@ -22,72 +22,121 @@ def edit
   def update
     update_params = match_contacts_params
     saved = @match_contacts.update(update_params)
-    unless request.xhr?
-      if saved
-        flash[:notice] = "Match Contacts updated"
-        redirect_to match_path(@match)
-      else
-        raise @match_contacts.errors.full_messages.inspect
-        flash[:error] = "Please review the form problems below."
-        redirect_to :edit
-      end
+    return if request.xhr?
+
+    if saved
+      flash[:notice] = 'Match Contacts updated'
+      redirect_to match_path(@match)
+    else
+      raise @match_contacts.errors.full_messages.inspect
+      flash[:error] = 'Please review the form problems below.' # rubocop:disable Lint/UnreachableCode
+      redirect_to :edit
     end
   end
 
+  def reorder
+    return unless params[:join_contact]
+
+    order_number = 1 # Default to setting the contact as primary
+    order_number = params[:order_number].to_i if params[:order_number]
+    set_contact_order(params[:join_contact], order_number)
+    set_match
+    set_match_contacts
+    # byebug
+    return if request.xhr?
+
+    redirect_to match_path(@match)
+  end
+
   private
 
-    def match_scope
-      ClientOpportunityMatch.all
+  def set_contact_order(update_contact, order_number = 1)
+    join_contact = match.client_opportunity_match_contacts.find(update_contact)
+    contact_type = nil
+    match_contact_methods.each do |type, _|
+      contact_type = type if join_contact.send(type)
     end
 
-    def set_match
-      @match = match_scope.find params[:match_id]
+    # We currently only want one contact per order number, so strip order status for any other contacts in the group
+    a_t = ClientOpportunityMatchContact.arel_table
+    contacts = @match.client_opportunity_match_contacts.where(a_t[contact_type].eq(true)).where(contact_order: order_number)
+    contacts.each do |contact|
+      contact.contact_order = nil
+      contact.save
     end
 
-    def set_current_contact
-      @current_contact = current_contact
-    end
+    # Set the selected contact as ordered contact
+    join_contact.contact_order = order_number
+    join_contact.save
+  end
 
-    def set_match_contacts
-      @match_contacts = @match.match_contacts
-    end
+  def match_contact_methods
+    @match_contact_methods ||= {
+      shelter_agency: :shelter_agency_contacts,
+      client: :client_contacts,
+      dnd_staff: :dnd_staff_contacts,
+      housing_subsidy_admin: :housing_subsidy_admin_contacts,
+      ssp: :ssp_contacts,
+      hsp: :hsp_contacts,
+      do: :do_contacts,
+    }
```

Comment:
Can we pull these from somewhere in the `ClientOpportunityMatch` or `ClientOpportunityMatchContact` models?  It makes me a bit nervous given that these have evolved over time.  I'd like to centralize that near where we define these types if not re-use some existing method.